lib/node_modules/@stdlib/math/base/special/imuldw/lib/main.js

```diff
@@ -0,0 +1,104 @@
+/**
+* @license Apache-2.0
+*
+* Copyright (c) 2018 The Stdlib Authors.
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*    http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+'use strict';
+
+// MODULES //
+
+var isnan = require( '@stdlib/math/base/assert/is-nan' );
+
+
+// VARIABLES //
+
+// Define a mask for the least significant 16 bits (low word): 65535 => 0x0000ffff => 00000000000000001111111111111111
+var LOW_WORD_MASK = 0x0000ffff>>>0; // asm type annotation
+
+
+// MAIN //
+
+/**
+* Performs multiplication of two signed 32-bit integers and returns an array of two signed 32-bit integers which represents the 64-bit signed integer product.
+*
+* @param {ArrayLikeObject} [out] - output array
+* @param {uinteger32} a - integer
+* @param {uinteger32} b - integer
+* @returns {ArrayLikeObject} output array
+*
+* @example
+* var v = imuldw( 0xAAAAAAAA, 0x55555555 );
+* // returns [ -477218589, 1908874354 ]
+*/
+function imuldw() {
+	var nargs;
+	var out;
+	var w1;
+	var w2;
+	var w3;
+	var ha;
+	var hb;
+	var la;
+	var lb;
+	var a;
+	var b;
+	var t;
+	var k;
+
+	nargs = arguments.length;
+	if ( nargs === 3 ) {
+		out = arguments[0];
+		a = arguments[1];
+		b = arguments[2];
+	} else {
+		out = new Array( 2 );
```

Comment:
```suggestion
		out = [ 0, 0 ];
```

Prefer initializing to values to ensure unboxed array elements (in this case).